File: src/test/java/com/flaghacker/uttt/common/JSONBoardTest.java
package com.flaghacker.uttt.common;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class JSONBoardTest
{
	@Test
	public void testEmpty()
	{
		Board board = new Board();
		Board result = JSONBoard.fromJSON(JSONBoard.toJSON(board));

		assertEquals(board, result);
	}

	@Test
	public void testRandom()
	{
		Board board = BoardTest.randomBoard(Util.loggedRandom(), 10);
		Board result = JSONBoard.fromJSON(JSONBoard.toJSON(board));

		assertEquals(board, result);
	}
}

File: src/test/com/flaghacker/sttt/JSONBoardTest.kt
package com.flaghacker.sttt

import com.flaghacker.sttt.common.Board
import com.flaghacker.sttt.common.JSONBoard
import com.flaghacker.sttt.common.toJSON
import org.junit.jupiter.api.Test
import java.util.*

class JSONBoardTest {
	@Test
	fun testEmpty() {
		val board = Board()
		val result = JSONBoard.fromJSON(board.toJSON())

		assertBoardEquals(board, result)
	}

	@Test
	fun testRandom() {
		val board = randomBoard(Random(0), 10)
		val result = JSONBoard.fromJSON(board.toJSON())

		assertBoardEquals(board, result)
	}
}